File: Week1Python/fizzbuzz_procedural.py
# Fizzbuzz

# Procedural 

print("Fizzbuzz in procedural")

# divisible_by(+number, +divisor)
# passig in number and divisor
def divisible_by(number, divisor):
    return number % divisor == 0

# fizzbuss(+limit)
def fizzbuzz(limit):
    for x in range(1, limit+1):
        if divisible_by(x, 15):
            print("Fizzbuzz")
        elif divisible_by(x, 3):
            print("Fizz")
        elif divisible_by(x, 5):
            print("Buzz")
        else:
            print(x)
            
fizzbuzz(100)

File: Week5Go/fizzbuzz_procedural.go
// Procedural fizzbuzz in Go

package main

import "fmt"

func divisibleBy(numerator, divisor int) bool {
	return numerator % divisor == 0
}

func fizzbuzz(limit int) {
	for i := 1; i <= limit; i++ {
		if divisibleBy(i, 15) {
			fmt.Println("FizzBuzz")
		} else if divisibleBy(i, 3) {
			fmt.Println("Fizz")
		} else if divisibleBy(i, 5) {
			fmt.Println("Buzz")
		} else {
			fmt.Println(i)
		}
	}
}

func main() {
	fizzbuzz(100)
}